src/main/java/co/omise/models/Charge.java

```diff
@@ -258,96 +268,211 @@ public void setSource(Source source) {
         @JsonProperty
         private String source;
 
-        public Create customer(String customer) {
+        @Override
+        protected String method() {
+            return POST;
+        }
+
+        @Override
+        protected HttpUrl path() {
+            return buildUrl(Endpoint.API, "charges");
+        }
+
+        @Override
+        protected RequestBody payload() throws IOException {
+            return serialize();
+        }
+
+        public CreateRequestBuilder customer(String customer) {
             this.customer = customer;
             return this;
         }
 
-        public Create card(String card) {
+        public CreateRequestBuilder card(String card) {
             this.card = card;
             return this;
         }
 
-        public Create amount(long amount) {
+        public CreateRequestBuilder amount(long amount) {
             this.amount = amount;
             return this;
         }
 
-        public Create currency(String currency) {
+        public CreateRequestBuilder currency(String currency) {
             this.currency = currency;
             return this;
         }
 
-        public Create description(String description) {
+        public CreateRequestBuilder description(String description) {
             this.description = description;
             return this;
         }
 
-        public Create metadata(Map<String, Object> metadata) {
-            // TODO: We should probably do an immutable copy here.
-            this.metadata = metadata;
+        public CreateRequestBuilder metadata(String key, Object value) {
+            HashMap<String, Object> tempMap = Maps.newHashMap();
+            tempMap.put(key, value);
+
+            this.metadata = new HashMap<>(tempMap);
```

Comment:
@Namiii I guess we can just do `this.metadata = tempMap` ? 😂

Reply:
That's true, was just goin g for consistency :)

Reply:
Changed it 👍